### PR TITLE
Added save_to_disk method to Deduper

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Added save_to_disk method to Deduperv in accordance with #84.

### CODE REVIEW

File: src/dfm/cleaning/deduper.py [hunk 1]
```diff
@@ -276,6 +277,70 @@ def deduplicate(
...
+        output_dir: Union[str, Path],
+        overwrite: bool = False,
+        store_mask_to_disk: bool = True,
+        store_lsh_cache_to_disk: bool = True,
+        store_config_to_disk: bool = True,
+    ) -> None:
+        """Save the Deduper to disk.
+
+        Args:
+            output_dir (str or Path, optional): The name of the output directory.
```
Comment (saattrupdan) [inline src/dfm/cleaning/deduper.py:291]:
```suggestion
            output_dir (str or Path, optional): 
                The name of the output directory.
```

File: src/dfm/cleaning/deduper.py [hunk 2]
```diff
@@ -359,43 +424,13 @@ def _deduplicate(
         # If the output file already exists then raise an error if `overwrite`
         # is False and otherwise delete the file
         if output_dir.exists():
```
Comment (saattrupdan) [inline src/dfm/cleaning/deduper.py:426]: This line is not needed, as it is dealt with in the `save_to_disk` method.

**Review by saattrupdan — CHANGES_REQUESTED** [review]
Looks good! Aside from the small comments below, the `save_to_disk` should now be used in the script instead of the normal saving blocks.

File: src/dfm/cleaning/deduper.py [hunk 3]
```diff
@@ -276,6 +277,70 @@ def deduplicate(
...
+                "the files. If you are loading an existing "
+                "Deduper from the directory then the previous "
+                "config, mask and LSH cache will still will "
+                "not be lost and will be stored in the directory."
+            )
+        elif output_dir.exists() and overwrite:
+            # Delete the output directory
+            shutil.rmtree(output_dir)
+
+        output_dir.mkdir(parents=True, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/dfm/cleaning/deduper.py:322]: I think python 3.7 is a reasonable assumption

[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/dfm/cleaning/deduper.py:322]: I works for python 3.7+ i think that is a reasonable assumption?

File: src/dfm/cleaning/deduper.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply src/dfm/cleaning/deduper.py:426]: It shouldn't always save to disk here. E.g. it should save an empty deduper.

File: src/dfm/cleaning/deduper.py [hunk 4]
```diff
@@ -359,43 +424,13 @@ def _deduplicate(
...
-                )
+            self.save_to_disk(
+                output_dir,
+                overwrite,
+                store_mask_to_disk,
+                store_lsh_cache_to_disk,
+                store_config_to_disk,
+            )
 
         # Create the output directory
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/dfm/cleaning/deduper.py:435]: See comment above.

**Comment by KennethEnevoldsen** [comment]
> Looks good! Aside from the small comments below, the save_to_disk should now be used in the script instead of the normal saving blocks.

if we want to replace the normal saving blocks with save_to_disk we will have to remove the folder deletion from save_to_disk otherwise it deletes the mask or raises an error.

File: src/dfm/cleaning/deduper.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (saattrupdan) [reply src/dfm/cleaning/deduper.py:323]: Ah, apparently [it was introduced in Python 3.5](https://docs.python.org/3.9/library/pathlib.html#pathlib.Path.mkdir), so we're all good!

File: src/dfm/cleaning/deduper.py (hunk 2 above)
Comment (saattrupdan) [reply src/dfm/cleaning/deduper.py:427]: Ah right, good point! Can you change the comment above (lines 359-360) to explain this?

**Review by saattrupdan — CHANGES_REQUESTED** [review]

**Comment by saattrupdan** [comment]
> If we want to replace the normal saving blocks with save_to_disk we will have to remove the folder deletion from save_to_disk otherwise it deletes the mask or raises an error.

Good point! Maybe make the folder deletion an argument to `save_to_disk` in that case? Would just be nice to remove redundant code, as that might cause annoying bugs down the line, if we only change the saving mechanism in one of the two places.

**Comment by KennethEnevoldsen** [comment]
Agree that is probably easier to maintain.

**Review by saattrupdan — APPROVED** [review]
LGTM!

**Pull request merged**